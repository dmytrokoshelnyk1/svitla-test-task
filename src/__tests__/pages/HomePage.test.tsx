// @ts-nocheck
import React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

import Home from '@/app/page'

describe('BackButton component tests', () => {
  it('Renders correctly initial component', async () => {
    render(<Home />)
    const p = screen.getByText("Hello, it's a test app for interview")
    const p2 = screen.getByText('A list of available Cable types')
    const h2 = screen.getByText('Cable types')

    expect(p).toBeInTheDocument()
    expect(p2).toBeInTheDocument()
    expect(h2).toBeInTheDocument()
  })

  it('Renders with user logo', async () => {
    const { getByAltText } = await render(<Home />)

    const image = getByAltText('Dmytro Kosheknyk avatar')

    expect(image.src).toContain('author_logo.jpeg')
  })

  it('Should have a link to /cables page', async () => {
    const { getByText } = render(<Home />)
    const link = getByText('Cable types').closest('a')
    expect(link).toBeInTheDocument()
    expect(link.href).toContain('/cables')
  })

  it('renders correctly toMatchSnapshot', () => {
    const tree = renderer.create(<Home />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
