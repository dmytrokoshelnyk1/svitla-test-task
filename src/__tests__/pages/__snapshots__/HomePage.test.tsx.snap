// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton component tests renders correctly 1`] = `
<main
  className="pageContainer"
>
  <div
    className="description"
  >
    <p>
      Hello, it's a test app for interview
    </p>
    <div>
      <a
        href="https://github.com/dmytrokoshelnyk1/svitla-test-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        By Dmytro Koshelnyk
        <img
          alt="Dmytro Kosheknyk avatar"
          className="authorLogo"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height={24}
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fauthor_logo.jpeg&w=256&q=75"
          srcSet="/_next/image?url=%2Fauthor_logo.jpeg&w=128&q=75 1x, /_next/image?url=%2Fauthor_logo.jpeg&w=256&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={100}
        />
      </a>
    </div>
  </div>
  <div
    className="content"
  >
    <div
      className="center"
    >
      <a
        className="card"
        href="/cables"
      >
        <h2>
          Cable types 
          <span>
            -&gt;
          </span>
        </h2>
        <p>
          A list of available Cable types
        </p>
      </a>
    </div>
  </div>
</main>
`;
